### ISSUE TITLE
Out of memory when sorting attributes

### ISSUE THREAD

**Issue by runningnak3d** [Lazza/RecuperaBit#25]
It appears that if you have a drive with a lot of files, when it goes to recreate the MFT, it runs out of memory.

I have a 4TB drive that had only 1 partition, but (rough guess) had hundreds of thousands of files.

Is there any debugging that I can add to try and figure out how much RAM will be needed? I have access to a box with 192gigs of RAM, but I would really like to figure out how much RAM will be needed for it to complete before I requisition the box.

**Comment by Lazza** [comment]
>  have a 4TB drive that had only 1 partition, but (rough guess) had hundreds of thousands of files.

Yes, large drives require a larger amount of RAM.

> Is there any debugging that I can add to try and figure out how much RAM will be needed?

I would like to ask you to tell me more precisely how many partitions are there and how many files are there. You can add this block inside `ntfs.py` before line 782:

```
        for address in partitioned_files:
            print address, '->', len(partitioned_files[address])
        return {}
```

Then run the program again. It will not reconstruct any partition but it will print how many files are in them. Thank you.

**Comment by ascattolini** [comment]
Terminated with "Killed", where should have been the output of how many files are in partitions?
Reporting the same thing:

[.....]
INFO:root:Found NTFS index record at sector 573391301
INFO:root:Found NTFS index record at sector 573391429
INFO:root:First scan completed
INFO:root:Parsing MFT entries
ERROR:root:Cannot handle multiple attribute $STANDARD_INFORMATION
ERROR:root:Problem parsing record on sector 2895556
ERROR:root:Cannot handle multiple attribute $STANDARD_INFORMATION
ERROR:root:Problem parsing record on sector 2895698
ERROR:root:Cannot handle multiple attribute $STANDARD_INFORMATION
ERROR:root:Problem parsing record on sector 1465432
ERROR:root:Cannot handle multiple attribute $STANDARD_INFORMATION
ERROR:root:Problem parsing record on sector 72507203
ERROR:root:Cannot handle multiple attribute $STANDARD_INFORMATION
ERROR:root:Problem parsing record on sector 72507285
ERROR:root:Cannot handle multiple attribute $STANDARD_INFORMATION
ERROR:root:Problem parsing record on sector 2808500
Killed

**Comment by runningnak3d** [comment]
I am guessing I didn't get the indentation correct.
```
INFO:root:First scan completed
INFO:root:Parsing MFT entries
INFO:root:Parsing INDX records
INFO:root:Reading boot sectors
6293504 ->Traceback (most recent call last):
  File "main.py", line 358, in <module>
    main()
  File "main.py", line 340, in main
    parts.update(scanner.get_partitions())
  File "/home/swango/dev/RecuperaBit/recuperabit/fs/ntfs.py", line 782, in get_partitions
    print address, '->', len(partitioned_files[address])
TypeError: 'NTFSPartition' has no length
```

**Comment by Lazza** [comment]
Guys could you try the fix proposed in #28?
